plugin/evm/vm_test.go

```diff
@@ -2268,6 +2301,110 @@ func TestTxAllowListSuccessfulTx(t *testing.T) {
 	}
 
 	require.Equal(t, signedTx0.Hash(), txs[0].Hash())
+
+	vm.clock.Set(reenableAllowlistTime.Add(time.Hour))
+
+	// Re-Submit a successful transaction
+	tx0 = types.NewTransaction(uint64(1), testEthAddrs[0], big.NewInt(1), 21000, big.NewInt(testMinGasPrice), nil)
+	signedTx0, err = types.SignTx(tx0, types.NewEIP155Signer(vm.chainConfig.ChainID), testKeys[0])
+	require.NoError(t, err)
+
+	errs = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx0})
+	require.NoError(t, errs[0])
+
+	// accept block to trigger upgrade
+	blk = issueAndAccept(t, issuer, vm)
+	newHead = <-newTxPoolHeadChan
+	require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()))
+	block = blk.(*chain.BlockWrapper).Block.(*Block).ethBlock
+
+	txs = block.Transactions()
+
+	blkState, err := vm.blockChain.StateAt(block.Root())
+	require.NoError(t, err)
+
+	// Check that address 0 is admin and address 1 is manager
+	role = txallowlist.GetTxAllowListStatus(blkState, testEthAddrs[0])
+	require.Equal(t, allowlist.AdminRole, role)
+	role = txallowlist.GetTxAllowListStatus(blkState, managerAddress)
+	require.Equal(t, allowlist.ManagerRole, role)
+
+	vm.clock.Set(vm.clock.Time().Add(2 * time.Second)) // add 2 seconds for gas fee to adjust
+	// Submit a successful transaction, should not throw an error because manager is activated
+	tx3 := types.NewTransaction(uint64(0), managerAddress, big.NewInt(1), 21000, big.NewInt(testMinGasPrice), nil)
+	signedTx3, err := types.SignTx(tx3, types.NewEIP155Signer(vm.chainConfig.ChainID), managerKey)
+	require.NoError(t, err)
+
+	vm.clock.Set(vm.clock.Time().Add(2 * time.Second)) // add 2 seconds for gas fee to adjust
+	errs = vm.txPool.AddRemotesSync([]*types.Transaction{signedTx3})
+	require.NoError(t, errs[0])
+
+	blk = issueAndAccept(t, issuer, vm)
+	newHead = <-newTxPoolHeadChan
+	require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()))
+
+	// Verify that the constructed block only has the whitelisted tx
+	block = blk.(*chain.BlockWrapper).Block.(*Block).ethBlock
+	txs = block.Transactions()
+
+	require.Len(t, txs, 1)
+	require.Equal(t, signedTx3.Hash(), txs[0].Hash())
+}
+
+func TestVerifyManagerConfig(t *testing.T) {
+	genesis := &core.Genesis{}
+	require.NoError(t, genesis.UnmarshalJSON([]byte(genesisJSONSubnetEVM)))
+
+	// this manager role should not be activated because DUpgradeTimestamp is in the future
+	genesis.Config.GenesisPrecompiles = params.Precompiles{
+		txallowlist.ConfigKey: txallowlist.NewConfig(utils.NewUint64(0), testEthAddrs[0:1], nil, []common.Address{testEthAddrs[1]}),
+	}
+
+	genesisJSON, err := genesis.MarshalJSON()
+	require.NoError(t, err)
+
+	vm := &VM{}
+	ctx, dbManager, genesisBytes, issuer, _ := setupGenesis(t, string(genesisJSON))
+	err = vm.Initialize(
+		context.Background(),
+		ctx,
+		dbManager,
+		genesisBytes,
+		[]byte(""),
+		[]byte(""),
+		issuer,
+		[]*commonEng.Fx{},
+		nil,
+	)
+	require.ErrorContains(t, err, "cannot add managers before DUpgrade")
```

Comment:
could we make this a defined error and use that (or at least its string representation) to avoid repeating this string?

Reply:
good call